ypy_websocket/websocket_provider.py

```diff
@@ -100,6 +109,9 @@ async def _run(self):
         async for message in self._websocket:
             if message[0] == YMessageType.SYNC:
                 await process_sync_message(message[1:], self._ydoc, self._websocket, self.log)
+                if self._synced is not None:
+                    self._synced.set()
+                    self._synced = None
```

Comment:
Synchronization is done after a `SYNC_STEP2` message is processed.
Also, I don't understand why you set `_synced` back to `None`. Shouldn't it be just `self.synced.set()`? Something like:
```suggestion
                if message[1] == YSyncMessageType.SYNC_STEP2:
                    self.synced.set()
```

Reply:
For my intended use case, either will work, but I figured that to mirror the flexibility of the JS (which IIUC can be triggered on _every_ sync), the semantics should be that the event fire on the next synchronization even if the initial sync has already happened. This way if you wanted to listen for each synchronization in a loop, you could, but it's still a useful interface if you only care about the first sync event.

I'm not committed to this, I would also be fine with the event firing only after the initial synchronization.

Reply:
Why would there be multiple synchronizations?